spec/migrations/20200607025146_add_ancestry_to_vm_spec.rb

```diff
@@ -0,0 +1,189 @@
+require_migration
+
+describe AddAncestryToVm do
+  let(:rel_stub) { migration_stub(:Relationship) }
+  let(:vm_stub) { migration_stub :VmOrTemplate }
+  let(:root) { vm_stub.create! }
+  let(:vm) { vm_stub.create! }
+  let(:vm2) { vm_stub.create! }
+  let(:vm3) { vm_stub.create! }
+  let(:vm4) { vm_stub.create! }
+  let(:vm5) { vm_stub.create! }
+  let(:vm6) { vm_stub.create! }
+  let(:vm7) { vm_stub.create! }
+  let(:vm8) { vm_stub.create! }
+
+  migration_context :up do
+    context "single parent/child rel" do
+      it 'updates ancestry' do
+        parent_rel = rel_stub.create!(:relationship => 'genealogy', :ancestry => nil, :resource_type => 'VmOrTemplate', :resource_id => root.id)
+        rel_stub.create!(:relationship => 'genealogy', :ancestry => ancestry_for(parent_rel), :resource_type => 'VmOrTemplate', :resource_id => vm.id)
+
+        migrate
+
+        expect(vm.reload.ancestry).to eq(ancestry_for(root))
+        expect(root.reload.ancestry).to eq(nil)
+        expect(rel_stub.count).to eq(0)
+      end
+    end
+
+    context "tree with depth 3" do
+      #         parent
+      #         child
+      #   second_level_child
+
+      it 'updates ancestry' do
+        parent_rel = rel_stub.create!(:relationship => 'genealogy', :ancestry => nil, :resource_type => 'VmOrTemplate', :resource_id => root.id)
+        child_rel = rel_stub.create!(:relationship => 'genealogy', :ancestry => ancestry_for(parent_rel), :resource_type => 'VmOrTemplate', :resource_id => vm.id)
+        rel_stub.create!(:relationship => 'genealogy', :ancestry => ancestry_for(child_rel, parent_rel), :resource_type => 'VmOrTemplate', :resource_id => vm2.id)
```

Comment:
They say comments are a code smell.
The same folks also say to comment your code.
Sigh.

So the only way to do that is to create self documenting code.

https://github.com/ManageIQ/manageiq/blob/master/spec/models/mixins/relationship_mixin_spec.rb#L3-L7

Sorry, just remembered that code when I was typing the last response. It was amusing that the code was for relationship mixin, since we are ... yea

If you look at the evolution from your first version to the second version, and you extrapulate, you'll see how someone ended up here.